task6/task6.py

```diff
@@ -0,0 +1,59 @@
+# Задача-1
+# Из текстового файла удалить все слова, содержащие от трех до пяти символов,
+# но при этом из каждой строки должно быть удалено только четное количество таких слов.
+import re
+
+
+def diff(first, second):
+    """ This function finds the difference between two list with keeping order."""
+    second = set(second)
+    return [item for item in first if item not in second]
+
+
+def delete_words_by_length(file, low_b=3, up_b=5):
+    """ This function deletes words in the text by given length and overwrites the file.
+    Hereinafter, the word 'puncs' will mean punctuation marks in the text.
+    Speciality: function deletes an even number of words on each line.
+    For the odd case it will overwrite the last word in a line.
+    :param file: File which needs to be changed and overwritten.
+    :param low_b: Lower bound of word's length you want to delete.
+    :param up_b: Upper bound... (the same)
+    """
+    with open(file, "r+") as f:
```

Comment:
все ок, только глядя на код, хочется его разделить на более просты функции.. например(можно больше):

```
def get_new_line(line, low_b, up_b):
    words = re.findall(r"[\w']+|[.,!?;:]", line)  # get a list of word including puncs as individual items
    short_words = [word for word in words if low_b <= len(word) <= up_b]
    if len(short_words) % 2 == 1:
        del short_words[-1:]

    difference = diff(words, short_words)
    new_line = " ".join(difference)
    return short_words


def delete_words_by_length(file, low_b=3, up_b=5):
    """ This function deletes words in the text by given length and overwrites the file.
    Hereinafter, the word 'puncs' will mean punctuation marks in the text.
    Speciality: function deletes an even number of words on each line.
    For the odd case it will overwrite the last word in a line.
    :param file: File which needs to be changed and overwritten.
    :param low_b: Lower bound of word's length you want to delete.
    :param up_b: Upper bound... (the same)
    """
    with open(file, "r+") as f:
        new_text = ""
        for line in f:
            new_line = get_new_line(line, low_b, up_b)
            corr_spaces = re.sub(r"\s([.,!?;:](?:\s|$))", r"\1", new_line)  # getting correct spaces before puncs
            new_text = "{}{}\n".format(new_text, corr_spaces)
        f.seek(0)
        f.truncate()
        f.write(new_text)

    delete_words_by_length("delete3-5.txt")
```